tests/integration/test_repl.py

```diff
@@ -402,3 +402,35 @@ def test_pipe_error(environment, couch_server):
         # environment.output_stream should be reverted to its original state
         # if anything in the pipe fails
         assert environment.output_stream is f
+
+
+def test_du_command_no_database_selected(environment, couch_server):
+    couch_server.create('test')
+    _, file_path = tempfile.mkstemp()
+    with io.open(file_path, 'w') as f:
```

Comment:
For normal output tests, the default for the environment fixture should work (which is backed by StringIO).
I used a temp file for testing pipes because StringIO cannot be used as stdin to other processes.